From: John Langford <jl@hunch.net>
Subject: bugfix for parallel code

--- a/vowpalwabbit/accumulate.cc
+++ b/vowpalwabbit/accumulate.cc
@@ -86,13 +86,13 @@ void accumulate_weighted_avg(vw& all, st
   float* local_weights = new float[length];
 
   for(uint32_t i = 0;i < length;i++) 
-    local_weights[i] = sqrt(weights[stride*i+1]*weights[stride*i+1]-1);
+    local_weights[i] = sqrt(weights[stride*i+1]*weights[stride*i+1]);
   
   all_reduce(local_weights, length, master_location, all.unique_id, all.total, all.node, all.socks);
 
   for(uint32_t i = 0;i < length;i++) 
     if(local_weights[i] > 0) {
-      float ratio = sqrt(weights[stride*i+1]*weights[stride*i+1]-1)/local_weights[i];
+      float ratio = sqrt(weights[stride*i+1]*weights[stride*i+1])/local_weights[i];
       weights[stride*i] *= ratio;
       weights[stride*i+1] *= ratio;
     }
